Dockerfile.arm64

```diff
@@ -0,0 +1,158 @@
+RUN apt-get update && apt-get install -y \
+  musl-dev \
+  musl-tools \
+  file \
+  git \
+  openssh-client \
+  make \
+  cmake \
+  g++ \
+  curl \
+  pkgconf \
+  ca-certificates \
+  xutils-dev \
+  libssl-dev \
+  libpq-dev \
+  automake \
+  autoconf \
+  libtool \
+  libprotobuf-dev \
+  unzip \
+  --no-install-recommends && \
+  rm -rf /var/lib/apt/lists/*
+
+# Install rust using rustup
+ARG CHANNEL
+ENV RUSTUP_VER="1.26.0" \
+    RUST_ARCH="aarch64-unknown-linux-gnu" \
+    CARGO_REGISTRIES_CRATES_IO_PROTOCOL=sparse
+
+RUN curl "https://static.rust-lang.org/rustup/archive/${RUSTUP_VER}/${RUST_ARCH}/rustup-init" -o rustup-init && \
+    chmod +x rustup-init && \
+    ./rustup-init -y --default-toolchain ${CHANNEL} --profile minimal --no-modify-path && \
+    rm rustup-init && \
+    ~/.cargo/bin/rustup target add aarch64-unknown-linux-musl
+
+# Allow non-root access to cargo
+RUN chmod a+X /root
+
+# Convenience list of versions and variables for compilation later on
+# This helps continuing manually if anything breaks.
+ENV SSL_VER="1.1.1w" \
+    CURL_VER="8.6.0" \
+    ZLIB_VER="1.3.1" \
+    PQ_VER="11.12" \
+    SQLITE_VER="3450100" \
+    PROTOBUF_VER="25.2" \
+    CC=musl-gcc \
+    PREFIX=/musl \
+    PATH=/usr/local/bin:/root/.cargo/bin:$PATH \
+    PKG_CONFIG_PATH=/usr/local/lib/pkgconfig \
+    LD_LIBRARY_PATH=$PREFIX
+
+# Install a more recent release of protoc (protobuf-compiler in jammy is 4 years old and misses some features)
+RUN cd /tmp && \
+    curl -sSL https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_VER}/protoc-${PROTOBUF_VER}-linux-aarch_64.zip -o protoc.zip && \
+    unzip protoc.zip && \
+    cp bin/protoc /usr/bin/protoc && \
+    rm -rf *
+
+# Set up a prefix for musl build libraries, make the linker's job of finding them easier
+# Primarily for the benefit of postgres.
+# Lastly, link some linux-headers for openssl 1.1 (not used herein)
+RUN mkdir $PREFIX && \
+    echo "$PREFIX/lib" >> /etc/ld-musl-aarch64.path && \
+    ln -s /usr/include/aarch64-linux-gnu/asm /usr/include/aarch64-linux-musl/asm && \
+    ln -s /usr/include/asm-generic /usr/include/aarch64-linux-musl/asm-generic && \
+    ln -s /usr/include/linux /usr/include/aarch64-linux-musl/linux
+
+# Build zlib (used in openssl and pq)
+RUN curl -sSL https://zlib.net/zlib-$ZLIB_VER.tar.gz | tar xz && \
+    cd zlib-$ZLIB_VER && \
+    CC="musl-gcc -fPIC -pie" LDFLAGS="-L$PREFIX/lib" CFLAGS="-I$PREFIX/include" ./configure --static --prefix=$PREFIX && \
+    make -j$(nproc) && make install && \
+    cd .. && rm -rf zlib-$ZLIB_VER
+
+# Build openssl (used in curl and pq)
+# Would like to use zlib here, but can't seem to get it to work properly
+# TODO: fix so that it works
+RUN curl -sSL https://www.openssl.org/source/openssl-$SSL_VER.tar.gz | tar xz && \
+    cd openssl-$SSL_VER && \
+    CFLAGS="-mno-outline-atomics" ./Configure no-zlib no-shared -fPIC --prefix=$PREFIX --openssldir=$PREFIX/ssl linux-aarch64 && \
```

Comment:
comparing amd64 and arm64:
```diff
-    CFLAGS="-mno-outline-atomics" ./Configure no-zlib no-shared -fPIC --prefix=$PREFIX --openssldir=$PREFIX/ssl linux-aarch64 && \
+    ./Configure no-zlib no-shared -fPIC --prefix=$PREFIX --openssldir=$PREFIX/ssl linux-x86_64 && \
```
this is the other big change, do you remember what was the deal with this?

Reply:
hehe ok. sounds like a standard wtf style google session then.

Reply:
yup...